pkg/util/progressbar.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file implements a go-getter wrapper for cheggaaa progress bar
+
+// based on:
+// https://github.com/hashicorp/go-getter/blob/master/cmd/go-getter/progress_tracking.go
+package util
+
+import (
+	"io"
+	"path/filepath"
+	"sync"
+
+	"github.com/cheggaaa/pb"
+	"github.com/golang/glog"
+	"github.com/hashicorp/go-getter"
+)
+
+var defaultProgressBar getter.ProgressTracker = &progressBar{}
+
+type progressBar struct {
+	lock sync.Mutex
+	pool *pb.Pool
+	pbs  int
+}
+
+// TrackProgress instantiates a new progress bar that will
+// display the progress of stream until closed.
+// total can be 0.
+func (cpb *progressBar) TrackProgress(src string, currentSize, totalSize int64, stream io.ReadCloser) io.ReadCloser {
```

Comment:
how do you feel about moving this to "out" package?

Reply:
I'd like to give download related code a home in a `download` package rather than `util`, but I felt that refactor was too big for a late PR blocker.

`out` is another possibility, but this implementation is specifically tied to go-getter.ProgessTracker, so I'd rather leave it nearer to the caller for convenience.

Reply:
ok fair enough